include/sys/util_macro.h

```diff
@@ -65,7 +65,7 @@ extern "C" {
  * @brief Bit mask with bits 0 through <tt>n-1</tt> (inclusive) set,
  * or 0 if @p n is 0.
  */
-#define BIT_MASK(n) (BIT(n) - 1UL)
+#define BIT_MASK(n) (BIT64(n) - 1UL)
```

Comment:
> I completely agree, I do not think we should do this at all. Instead, please add a new `BIT_MASK64()` that uses `BIT64`

In fact, there is already `BIT64_MASK` right below that does just that:
https://github.com/zephyrproject-rtos/zephyr/blob/a92ca1759e06ea4d1f4dd51fe01cd2b4693bfeff/include/sys/util_macro.h#L74

Reply:
Hi, @andyross @carlescufi @stephanosio 
Not sure I was clear enough, what I'm trying to fix is this code sample

uint32_t mask = BIT_MASK(32); /* 0xffffffff */

Are you suggesting to use this instead ?
uint32_t mask = BIT64_MASK(32); /* 0xffffffff */

Reply:
This fix is a legitimate warning workaround for the 32 bit version of the macro.  It just needs a cast on the final expression to prevent changing the API (effectively what the code as written does is, in fact, implement BIT64_MASK(), and that's not what you want.

Reply:
Ok thanks you all !